src/main/java/org/reactome/addlinks/fileprocessors/RGDFileProcessor.java

```diff
@@ -0,0 +1,53 @@
+package org.reactome.addlinks.fileprocessors;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class RGDFileProcessor extends FileProcessor{
```

Comment:
My comments for this class are similar to the `MGIFileProcessor` class.  It might be worth extracting some of the common logic out to a class that will be a child of `FileProcessor` and parent to `MGIFileProcessor`, `RGDFileProcessor`, and other similar classes.

Another option is to create a helper class that will contain the common logic and have it be an instance variable in MGI, RGD, and other file processor classes.  Then you can delegate the common work to the helper class.  This can sometimes be preferred to creating an inheritance hierarchy: https://www.javaworld.com/article/3409071/java-challenger-7-debugging-java-inheritance.html.  I'm not sure which is better in this case.  If these file processors make use of all of the `FileProcessor` class functionality and pass the "is a" test, inheritance may be better but otherwise composition may be preferrable.

Reply:
So its not possible to entirely abstract out the file processors due to the differences in the files they are processing. I moved some helper methods to the parent class that handles some of the common functionality.

Reply:
Do the moved helper methods apply to all the children of the parent class?  If not, you might want to consider creating a class outside the hierarchy and have it be used by the classes that need it.

Reply:
Is the paradigm for the parent class to have methods that are only ever always used by all classes? The reason I ask is that these are fairly generic helper methods that could probably be applied to other FileProcessors (but not all).

Reply:
Yes, a child class should never be inheriting a method from a parent that isn't applicable to it.

Reply:
"Never" seems a little too strict ;)
If the methods in the parent class _could_ be applied to _most_ of the child classes, and they make sense there (in the parent), conceptually, I don't think there's anything wrong. That being said, it could vary from case to case, I'd really have to see it to know for sure.

Reply:
For now I just moved the methods to the BiomartUtil class that already existed, so it wasn't too big of a deal